src/battleground/gameobjects/Bullet.js

```diff
@@ -0,0 +1,205 @@
+//@flow strict
+
+import Vec from '../../casus/blocks/Vec.js';
+import Seg from '../../geometry/Seg.js';
+import GameObject from '../GameObject.js';
+import {getImage} from '../ImageLoader.js';
+
+import type ImageDrawer from '../ImageDrawer.js';
+import type Battleground from '../Battleground.js';
+import type Tank from '../../tanks/Tank.js';
+import {createSmokeCloud, createElectricityParticle} from './Particle.js';
+
+type BulletType = 
+	'DEATH_RAY_BULLET' |
+	'GREEN_LASER' |
+	'GRENADE_BULLET' |
+	'GUN_BULLET' |
+	'RED_LASER' |
+	'MISSILE' |
+	'PLASMA_BLOB' |
+	'PULSE_LASER_PARTICLE' |
+	'SHOTGUN_BULLET'|
+	'LANCER_PARTICLE';
+
+type BulletStats = {
+	speed: number,
+	width: number,
+	lifetime: number,
+};
+
+const STATS_FOR_BULLET: {[BulletType]: BulletStats} = {
+	DEATH_RAY_BULLET: {
+		speed: 1.4,
+		width: 25,
+		lifetime: 100,
+	},
+	GREEN_LASER: {
+		speed: 7,
+		width: 40,
+		lifetime: 100,
+	},
+	GRENADE_BULLET: {
+		speed: 2,
+		width: 12,
+		lifetime: 15,
+	},
+	GUN_BULLET: {
+		speed: 2,
+		width: 13,
+		lifetime: 39,
+	},
+	RED_LASER: {
+		speed: 6,
+		width: 20,
+		lifetime: 100,
+	},
+	MISSILE: {
+		speed: 1.4,
+		width: 10,
+		lifetime: 100,
+	},
+	PLASMA_BLOB: {
+		speed: 1.2,
+		width: 20,
+		lifetime: 100,
+	},
+	PULSE_LASER_PARTICLE: {
+		speed: 0,
+		width: 150,
+		lifetime: 18,
+	},
+	SHOTGUN_BULLET: {
+		speed: 4,
+		width: 13,
+		lifetime: 10,
+	},
+	LANCER_PARTICLE: {
+		speed: 3,
+		width: 1,
+		lifetime: 0,
+	}
+};
+
+class Bullet extends GameObject {
+
+	rotation: number;
+	tankToIgnore: Tank;
```

Comment:
Is this supposed to be the user's tank that is firing the bullet?

Reply:
yeah

Reply:
(assuming you can't shoot yourself, which I thought was a fair assumption to make)